From 6ffef09e3cca8c66f88a7b7ac1fecf0f0ee94bc4 Mon Sep 17 00:00:00 2001
From: Tom Rix <trix@redhat.com>
Date: Tue, 18 Feb 2020 19:17:24 -0500
Subject: [PATCH] Add INTEL_FPGA_API_VERSION version to shared objects.

Most of the shared objects are of the form <name>.so.<INTEL_API_VERSION>
In this case <name>.so.1.4.0 with standard symlinks <name>.so.1 and
<name>.so.  For those that do not, provide the versioning to make them
consistent.

Do not version *.so's marked as modules.
---
 libopae/CMakeLists.txt    | 3 +++
 libopaecxx/CMakeLists.txt | 5 +++++
 2 files changed, 8 insertions(+)

diff --git a/libopaecxx/CMakeLists.txt b/libopaecxx/CMakeLists.txt
index dc0cb429..06ae1a60 100644
--- a/usr/libopaecxx/CMakeLists.txt
+++ b/usr/libopaecxx/CMakeLists.txt
@@ -64,6 +64,11 @@ add_executable(hello_cxxcore samples/hello_fpga-1.cpp)
 target_link_libraries(hello_cxxcore opae-c ${CMAKE_THREAD_LIBS_INIT} opae-cxx-core )
 
 
+# Target properties
+set_target_properties(opae-cxx-core PROPERTIES
+  VERSION ${INTEL_FPGA_API_VERSION}
+  SOVERSION ${INTEL_FPGA_API_VER_MAJOR})
+
 # Define headers for this library. PUBLIC headers are used for
 # compiling the library, and will be added to consumers' build
 # paths. Keep current directory private.
-- 
2.24.1

